buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -503,7 +504,9 @@ public ByteBuf getBytes(int index, ByteBuf dst, int length) {
 
     @Override
     public CharSequence getCharSequence(int index, int length, Charset charset) {
-        // TODO: We could optimize this for UTF8 and US_ASCII
+        if (CharsetUtil.US_ASCII.equals(charset) || CharsetUtil.ISO_8859_1.equals(charset)) {
+            return new AsciiString(ByteBufUtil.getBytes(this, index, length), false);
```

Comment:
Doesn't this have to be a copy?   Mutating the underlying buffer would modify the string.

Reply:
`ByteBufUtil.getBytes(...)` returns a fresh copy by default (there is also a "shared if possible" boolean param version but that's not being used here).

Reply:
@njhill maybe call the the other variant which takes a boolean to make it clear that it returns a copy ? Also a comment may help :)

Reply:
@normanmaurer done!